Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Confirm.hs

```diff
@@ -112,7 +113,8 @@ handler ::
 handler transporter req quote = do
   booking <- QRB.findById req.bookingId >>= fromMaybeM (BookingDoesNotExist req.bookingId.getId)
   now <- getCurrentTime
-  (riderDetails, isNewRider) <- getRiderDetails transporter.id req.customerMobileCountryCode req.customerPhoneNumber now
+  (riderDetails, isNewRider) <- getRiderDetails transporter.id req.customerMobileCountryCode req.customerPhoneNumber now req.nightSafetyCheck
+  unless isNewRider $ QRD.updateNightSafetyChecks riderDetails.id req.nightSafetyCheck
```

Comment:
In case of new rider we are already creating the riderDetails entity in DB in the same function. So night safety check will automatically be added there.